<launch>

	<arg name="rviz" default="true" />
	<arg name="autorun"   default="false"/>
    <arg name="scene_name" default="Scene_01"/>
    <arg name="data_path" default="/media/gtx/Gtx04/Collected_Dataset"/>
    <arg name="bag_file1"  default="$(arg data_path)/$(arg scene_name)/Localization/$(arg scene_name).bag"/>
	<arg name="bag_file2"  default="$(arg data_path)/$(arg scene_name)/Streetlight_Detection/$(arg scene_name)_Yolo.bag"/>

	<rosparam command="load" file="$(find night_voyager)/config/Config_$(arg scene_name).yaml" />

    <param name="display_ground_truth" type="bool" value="true" />
    <param name="ground_truth_path" type="string" value="$(arg data_path)/$(arg scene_name)/Ground_Truth/Ground_Truth.txt" />

    <param name="pcd_path" type="string" value="$(arg data_path)/$(arg scene_name)/Streetlight_Map/Lamps.txt" />
    <param name="virtual_center_path" type="string" value="$(arg data_path)/$(arg scene_name)/Streetlight_Map/Virtual_Centers.txt" />
    <param name="downsampled_pose_path" type="string" value="$(arg data_path)/$(arg scene_name)/Streetlight_Map/Downsampled_Poses.txt" />
    <param name="prior_pose_path" type="string" value="$(arg data_path)/$(arg scene_name)/Streetlight_Map/Selected_Poses.txt" />

    <node pkg="night_voyager" type="night_voyager_node" name="night_voyager_node" output="screen" /> 

	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find night_voyager)/rviz/Night_Voyager_test.rviz"/>
    <!-- <param name="/use_sim_time" value="true" /> -->
    <node required="$(arg autorun)" pkg="rosbag" type="play" name="rosbag_player"
          args="--clock $(arg bag_file1) $(arg bag_file2) -r 1.0 -s 0"
          launch-prefix="bash -c 'sleep 1; $0 $@' "/>

</launch>
